GoogleForCreators/web-stories-wp
# Tests: Improve PHPUnit Forward Compatibility

**PHPUnit 8.x and 9.x compatibility**

tl;dr: we need to update our PHP integration tests running against core so that they function properly (again) as needed.

WordPress core has moved to adopt PHPUnit polyfills for running its tests and at the same time runs tests using the latest possible PHPUnit version.

To make our test suite compatible, we need to:

* Update assertions so we don't use deprecated ones
* Define `WP_TESTS_PHPUNIT_POLYFILLS_PATH`
* Update PHPUnit to latest possible version on CI
* Update PHPUnit configuration to improve forward compatibility
* Update `Yoast/wp-test-utils` package

The PR also includes one small change related to PHP 8.1 compatibility, see #8520.

See https://core.trac.wordpress.org/ticket/46149 and some of the early individual commit messages for details.

See also https://core.trac.wordpress.org/ticket/53911 for when the PHPUnit changes were backported to older branches
